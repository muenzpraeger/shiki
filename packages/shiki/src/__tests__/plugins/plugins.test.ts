import { getHighlighter } from '../../index'
import pluginAsObject from './pluginAsObject'
import pluginAsFunction from './pluginAsFunction'
import pluginWithExplanationRequest from './pluginWithExplanations'
import pluginWithThemeRequest from './pluginWithTheme'
import pluginAlterEverythingWithValues from './pluginAlterEverythingValues'
import pluginAlterEverythingWithFunctions from './pluginAlterEverythingFunctions'

test('Loads without element modifications when no plugin is used', async () => {
  const highlighter = await getHighlighter({
    theme: 'nord',
    langs: ['js']
  })

  const out = highlighter.codeToHtml("console.log('shiki');", { lang: 'js' })

  expect(out).toMatchSnapshot()
})

test('Loads and applies a plugin that was returned as an object', async () => {
  const highlighter = await getHighlighter({
    theme: 'nord',
    langs: ['js']
  })

  highlighter.usePlugin(pluginAsObject)

  const out = highlighter.codeToHtml("console.log('shiki');", { lang: 'js' })

  expect(out).toMatchSnapshot()
})

test('Loads and applies a plugin that was returned as a function', async () => {
  const highlighter = await getHighlighter({
    theme: 'nord',
    langs: ['js']
  })

  highlighter.usePlugin(pluginAsFunction({ classToAdd: 'not-the-one-it-should-be' }))

  const out = highlighter.codeToHtml("console.log('shiki');", { lang: 'js' })

  expect(out).toMatchSnapshot()
})

test('Loads and applies a plugin that requests theme information', async () => {
  const highlighter = await getHighlighter({
    theme: 'nord',
    langs: ['js']
  })

  highlighter.usePlugin(pluginWithThemeRequest())

  const out = highlighter.codeToHtml("console.log('shiki');", { lang: 'js' })

  expect(out).toMatchSnapshot()
})

test('Loads and applies a plugin that requests token explanations', async () => {
  const highlighter = await getHighlighter({
    theme: 'nord',
    langs: ['js']
  })

  highlighter.usePlugin(pluginWithExplanationRequest())

  const out = highlighter.codeToHtml("console.log('shiki');", { lang: 'js' })

  expect(out).toMatchSnapshot()
})

test('Loads and applies a plugin that alters everything via values', async () => {
  const highlighter = await getHighlighter({
    theme: 'nord',
    langs: ['js']
  })

  highlighter.usePlugin(pluginAlterEverythingWithValues)

  const out = highlighter.codeToHtml("console.log('shiki');", { lang: 'js' })

  expect(out).toMatchSnapshot()
})

test('Loads and applies a plugin that alters everything via functions', async () => {
  const highlighter = await getHighlighter({
    theme: 'nord',
    langs: ['js']
  })

  highlighter.usePlugin(pluginAlterEverythingWithFunctions)

  const out = highlighter.codeToHtml("console.log('shiki');", { lang: 'js' })

  expect(out).toMatchSnapshot()
})

test('Loads and applies multiple plugins with overwriting duplicate keys', async () => {
  const highlighter = await getHighlighter({
    theme: 'nord',
    langs: ['js']
  })

  highlighter
    .usePlugin(pluginAlterEverythingWithValues)
    .usePlugin(pluginAlterEverythingWithFunctions)

  const out = highlighter.codeToHtml("console.log('shiki');", { lang: 'js' })

  expect(out).toMatchSnapshot()
})
