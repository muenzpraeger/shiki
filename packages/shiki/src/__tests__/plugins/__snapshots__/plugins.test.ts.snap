// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loads and applies a plugin that alters everything via functions 1`] = `"<pre class=\\"shiki nord funcPre1 funcPre2\\" style=\\"background-color: yellow; color: red;\\"  data-test=\\"funcPre\\"><code  data-test=\\"funcCode\\"><span class=\\"line funcLine1 funcLine2\\" style=\\"background-color: red; color: black;\\"  data-test=\\"funcLine\\"><span class=\\"funcToken1 funcToken2\\" style=\\"color: red; background-color: grey;\\"  data-test=\\"funcToken\\">console</span><span class=\\"funcToken1 funcToken2\\" style=\\"color: red; background-color: grey;\\"  data-test=\\"funcToken\\">.</span><span class=\\"funcToken1 funcToken2\\" style=\\"color: red; background-color: grey;\\"  data-test=\\"funcToken\\">log</span><span class=\\"funcToken1 funcToken2\\" style=\\"color: red; background-color: grey;\\"  data-test=\\"funcToken\\">(</span><span class=\\"funcToken1 funcToken2\\" style=\\"color: red; background-color: grey;\\"  data-test=\\"funcToken\\">&#39;</span><span class=\\"funcToken1 funcToken2\\" style=\\"color: red; background-color: grey;\\"  data-test=\\"funcToken\\">shiki</span><span class=\\"funcToken1 funcToken2\\" style=\\"color: red; background-color: grey;\\"  data-test=\\"funcToken\\">&#39;</span><span class=\\"funcToken1 funcToken2\\" style=\\"color: red; background-color: grey;\\"  data-test=\\"funcToken\\">)</span><span class=\\"funcToken1 funcToken2\\" style=\\"color: red; background-color: grey;\\"  data-test=\\"funcToken\\">;</span></span></code></pre>"`;

exports[`Loads and applies a plugin that alters everything via values 1`] = `"<pre class=\\"shiki nord valuePre1 valuePre2\\" style=\\"background-color: red; color: blue;\\"  data-test=\\"valuePre\\"><code  data-test=\\"valueCode\\"><span class=\\"line valueLine1 valueLine2\\" style=\\"background-color: yellow; color: green;\\"  data-test=\\"valueLine\\"><span class=\\"valueToken1 valueToken2\\" style=\\"color: yellow; background-color: green;\\"  data-test=\\"valueToken\\">console</span><span class=\\"valueToken1 valueToken2\\" style=\\"color: yellow; background-color: green;\\"  data-test=\\"valueToken\\">.</span><span class=\\"valueToken1 valueToken2\\" style=\\"color: yellow; background-color: green;\\"  data-test=\\"valueToken\\">log</span><span class=\\"valueToken1 valueToken2\\" style=\\"color: yellow; background-color: green;\\"  data-test=\\"valueToken\\">(</span><span class=\\"valueToken1 valueToken2\\" style=\\"color: yellow; background-color: green;\\"  data-test=\\"valueToken\\">&#39;</span><span class=\\"valueToken1 valueToken2\\" style=\\"color: yellow; background-color: green;\\"  data-test=\\"valueToken\\">shiki</span><span class=\\"valueToken1 valueToken2\\" style=\\"color: yellow; background-color: green;\\"  data-test=\\"valueToken\\">&#39;</span><span class=\\"valueToken1 valueToken2\\" style=\\"color: yellow; background-color: green;\\"  data-test=\\"valueToken\\">)</span><span class=\\"valueToken1 valueToken2\\" style=\\"color: yellow; background-color: green;\\"  data-test=\\"valueToken\\">;</span></span></code></pre>"`;

exports[`Loads and applies a plugin that requests theme information 1`] = `"<pre class=\\"shiki nord\\" style=\\"background-color: #3b4252;\\"><code><span class=\\"line\\"><span style=\\"color: #D8DEE9;\\">console</span><span style=\\"color: #ECEFF4;\\">.</span><span style=\\"color: #88C0D0;\\">log</span><span style=\\"color: #D8DEE9FF;\\">(</span><span style=\\"color: #ECEFF4;\\">&#39;</span><span style=\\"color: #A3BE8C;\\">shiki</span><span style=\\"color: #ECEFF4;\\">&#39;</span><span style=\\"color: #D8DEE9FF;\\">)</span><span style=\\"color: #81A1C1;\\">;</span></span></code></pre>"`;

exports[`Loads and applies a plugin that requests token explanations 1`] = `"<pre class=\\"shiki nord\\" style=\\"background-color: #2e3440ff;\\"><code><span class=\\"line\\"><span style=\\"color: #D8DEE9; background-color: blue;\\">console</span><span style=\\"color: #ECEFF4; background-color: red;\\">.</span><span style=\\"color: #88C0D0; background-color: red;\\">log</span><span style=\\"color: #D8DEE9FF; background-color: red;\\">(</span><span style=\\"color: #ECEFF4; background-color: red;\\">&#39;</span><span style=\\"color: #A3BE8C; background-color: red;\\">shiki</span><span style=\\"color: #ECEFF4; background-color: red;\\">&#39;</span><span style=\\"color: #D8DEE9FF; background-color: red;\\">)</span><span style=\\"color: #81A1C1; background-color: red;\\">;</span></span></code></pre>"`;

exports[`Loads and applies a plugin that was returned as a function 1`] = `"<pre class=\\"shiki nord plugin-as-function\\" style=\\"background-color: #2e3440ff;\\"><code><span class=\\"line\\"><span style=\\"color: #D8DEE9;\\">console</span><span style=\\"color: #ECEFF4;\\">.</span><span style=\\"color: #88C0D0;\\">log</span><span style=\\"color: #D8DEE9FF;\\">(</span><span style=\\"color: #ECEFF4;\\">&#39;</span><span style=\\"color: #A3BE8C;\\">shiki</span><span style=\\"color: #ECEFF4;\\">&#39;</span><span style=\\"color: #D8DEE9FF;\\">)</span><span style=\\"color: #81A1C1;\\">;</span></span></code></pre>"`;

exports[`Loads and applies a plugin that was returned as an object 1`] = `"<pre class=\\"shiki nord plugin-as-object\\" style=\\"background-color: #2e3440ff;\\"><code><span class=\\"line\\"><span style=\\"color: #D8DEE9;\\">console</span><span style=\\"color: #ECEFF4;\\">.</span><span style=\\"color: #88C0D0;\\">log</span><span style=\\"color: #D8DEE9FF;\\">(</span><span style=\\"color: #ECEFF4;\\">&#39;</span><span style=\\"color: #A3BE8C;\\">shiki</span><span style=\\"color: #ECEFF4;\\">&#39;</span><span style=\\"color: #D8DEE9FF;\\">)</span><span style=\\"color: #81A1C1;\\">;</span></span></code></pre>"`;

exports[`Loads and applies multiple plugins with overwriting duplicate keys 1`] = `"<pre class=\\"shiki nord valuePre1 valuePre2 funcPre1 funcPre2\\" style=\\"background-color: yellow; color: red;\\"  data-test=\\"funcPre\\"><code  data-test=\\"funcCode\\"><span class=\\"line valueLine1 valueLine2 funcLine1 funcLine2\\" style=\\"background-color: red; color: black;\\"  data-test=\\"funcLine\\"><span class=\\"valueToken1 valueToken2 funcToken1 funcToken2\\" style=\\"color: red; background-color: grey;\\"  data-test=\\"funcToken\\">console</span><span class=\\"valueToken1 valueToken2 funcToken1 funcToken2\\" style=\\"color: red; background-color: grey;\\"  data-test=\\"funcToken\\">.</span><span class=\\"valueToken1 valueToken2 funcToken1 funcToken2\\" style=\\"color: red; background-color: grey;\\"  data-test=\\"funcToken\\">log</span><span class=\\"valueToken1 valueToken2 funcToken1 funcToken2\\" style=\\"color: red; background-color: grey;\\"  data-test=\\"funcToken\\">(</span><span class=\\"valueToken1 valueToken2 funcToken1 funcToken2\\" style=\\"color: red; background-color: grey;\\"  data-test=\\"funcToken\\">&#39;</span><span class=\\"valueToken1 valueToken2 funcToken1 funcToken2\\" style=\\"color: red; background-color: grey;\\"  data-test=\\"funcToken\\">shiki</span><span class=\\"valueToken1 valueToken2 funcToken1 funcToken2\\" style=\\"color: red; background-color: grey;\\"  data-test=\\"funcToken\\">&#39;</span><span class=\\"valueToken1 valueToken2 funcToken1 funcToken2\\" style=\\"color: red; background-color: grey;\\"  data-test=\\"funcToken\\">)</span><span class=\\"valueToken1 valueToken2 funcToken1 funcToken2\\" style=\\"color: red; background-color: grey;\\"  data-test=\\"funcToken\\">;</span></span></code></pre>"`;

exports[`Loads without element modifications when no plugin is used 1`] = `"<pre class=\\"shiki nord\\" style=\\"background-color: #2e3440ff;\\"><code><span class=\\"line\\"><span style=\\"color: #D8DEE9\\">console</span><span style=\\"color: #ECEFF4\\">.</span><span style=\\"color: #88C0D0\\">log</span><span style=\\"color: #D8DEE9FF\\">(</span><span style=\\"color: #ECEFF4\\">&#39;</span><span style=\\"color: #A3BE8C\\">shiki</span><span style=\\"color: #ECEFF4\\">&#39;</span><span style=\\"color: #D8DEE9FF\\">)</span><span style=\\"color: #81A1C1\\">;</span></span></code></pre>"`;
